### PR TITLE
[MULTIARCH-3240] IBM Z installation updates 4.13

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s): 4.13 +

Issue: https://issues.redhat.com/browse/MULTIARCH-3240

Link to docs preview: 
- IBM Z: 
  - https://57055--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z.html#minimum-ibm-z-system-requirements_installing-ibm-z
  - https://57055--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-kvm.html#host-machine-resource-requirements_installing-ibm-z-kvm
  - https://57055--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-kvm.html#installing-rhcos-using-ibm-secure-execution_installing-ibm-z-kvm

- RHCOS based on 9.2 change for all arches: 
  - https://57055--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-machine-requirements_installing-bare-metal

- RHEL version changes x86: 
  - https://57055--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal.html#installation-minimum-resource-requirements_installing-bare-metal

QE review: 
- IBM Z: Holger Wolf
- RHCOS/RHEL: Mike Nguyen
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by jschintag — APPROVED** [review]
lgtm

**Comment by SNiemann15** [comment]
@nikita-dubrovskii Can you please review the SE section.

File: modules/ibm-z-secure-execution.adoc [hunk 1]
```diff
@@ -101,4 +98,44 @@ base64 <your-hostkey>.crt
...
+====
+.. Fetch the public GPG key for the `secex-qemu.qcow2` image and encrypt the Ignition config with the key, by running the following command:
++
+[source,terminal]
+----
+gpg --recipient-file /path/to/ignition.gpg.pub --yes --output /path/to/config.ign.gpg --armor --encrypt /path/to/config.ign
+----
++
+The output shows that encryption was successful:
++
```
Comment (nikita-dubrovskii) [inline modules/ibm-z-secure-execution.adoc:117]: Output is available only when `--verbose` flag is specified:
```
$ gpg --recipient-file /rhcos/builds/latest/s390x/rhcos-413.92.202303070857-0-ignition-secex-key.gpg.pub --yes --output config.ign.gpg --verbose --armor --encrypt ~/se.ign 
gpg: reading from '/tmp/se.ign'
gpg: writing to 'config.ign.gpg'
gpg: RSA/AES256.OCB encrypted for: "D7A3078D077F1CDD Secure Execution (secex) 413.92.202303070857-0"
```

File: modules/ibm-z-secure-execution.adoc [hunk 2]
```diff
@@ -101,4 +98,44 @@ base64 <your-hostkey>.crt
...
+.Example output
```
Comment (nikita-dubrovskii) [inline modules/ibm-z-secure-execution.adoc:118]: this output was taken from CoreOS `firstboot` and shows whether or not `config.ign.gpg` was decrypted

File: modules/ibm-z-secure-execution.adoc [hunk 3]
```diff
@@ -101,4 +98,44 @@ base64 <your-hostkey>.crt
...
+====
+To achieve the desired functionality, Ignition logging and local login are disabled by default when running IBM Secure Execution.
+====
+.. Fetch the public GPG key for the `secex-qemu.qcow2` image and encrypt the Ignition config with the key, by running the following command:
++
+[source,terminal]
+----
+gpg --recipient-file /path/to/ignition.gpg.pub --yes --output /path/to/config.ign.gpg --armor --encrypt /path/to/config.ign
+----
++
```
Comment (nikita-dubrovskii) [inline modules/ibm-z-secure-execution.adoc:115]: before starting the VM `config.ign.gpg` should be attached with new `serial=ignition_crypted` :
```
-drive if=none,id=ignition,format=raw,file=config.ign.gpg,readonly=on -device virtio-blk,serial=ignition_crypted,drive=ignition
```

File: modules/ibm-z-secure-execution.adoc [hunk 4]
```diff
@@ -101,4 +98,44 @@ base64 <your-hostkey>.crt
 +
```
Comment (nikita-dubrovskii) [inline modules/ibm-z-secure-execution.adoc:98]: I haven't found this, but would suggest to add note that user has to enable `ProtVirt` qemu [feature](https://qemu.readthedocs.io/en/latest/system/s390x/protvirt.html):
`-object s390-pv-guest,id=pv0 -machine confidential
-guest-support=pv0`

Comment (jschintag) [reply modules/ibm-z-secure-execution.adoc:98]: I think we had this thread about how, for different reasons, we removed all qemu flags from docs? So maybe we want to include that all io-devices need to use iommu?

**Comment by nikita-dubrovskii** [comment]
LGTM

**Review by nikita-dubrovskii — APPROVED** [review]

**Comment by SNiemann15** [comment]
/assign @mike-nguyen Can you please approve the RHCOS/RHEL changes in this PR. 

cc: @marrusl

**Comment by clapolt** [comment]
LGTM

**Comment by holgwolf** [comment]
LGTM

**Comment by ousleyp** [comment]
Hi @SNiemann15, it looks like there is a merge conflict on this branch. Please resolve the merge conflict before reapplying the peer-review-needed label. Thanks!

**Review by snarayan-redhat — COMMENTED** [review]
Looks good overall. Few suggestions.

File: modules/ibm-z-secure-execution.adoc [hunk 5]
```diff
@@ -101,4 +98,49 @@ base64 <your-hostkey>.crt
...
 Compared to guests not running IBM Secure Execution, the first boot of the machine is longer because the entire image is encrypted with a randomly generated LUKS passphrase before the Ignition phase.
 
+. Add Ignition protection
++
+To protect secrets that are stored in the Ignition config file from being read or even modified the   Ignition config file must be encrypted.
++
+[NOTE]
+====
+To achieve the desired functionality, Ignition logging and local login are disabled by default when running IBM Secure Execution.
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/ibm-z-secure-execution.adoc]: To be more accurate I changed to "security"

File: modules/installation-machine-requirements.adoc [hunk 6]
```diff
@@ -87,14 +87,14 @@ endif::ibm-z[]
 ====
 
 ifndef::ibm-z,ibm-power[]
-The bootstrap and control plane machines must use {op-system-first} as the operating system. However, the compute machines can choose between {op-system-first}, {op-system-base-full} 8.4, or {op-system-base} 8.5.
+The bootstrap and control plane machines must use {op-system-first} as the operating system. However, the compute machines can choose between {op-system-first}, {op-system-base-full} 8.6, or {op-system-base} 8.7.
```
Comment (mike-nguyen) [inline modules/installation-machine-requirements.adoc:90]: I'm not the person to review the bring your own RHEL.

File: modules/installation-machine-requirements.adoc [hunk 7]
```diff
@@ -87,14 +87,14 @@ endif::ibm-z[]
...
 endif::ibm-z,ibm-power[]
 ifdef::ibm-z,ibm-power[]
 The bootstrap, control plane, and compute machines must use {op-system-first} as the operating system.
 endif::ibm-z,ibm-power[]
 
 ifndef::openshift-origin[]
-Note that {op-system} is based on {op-system-base-full} 8 and inherits all of its hardware certifications and requirements.
+Note that {op-system} is based on {op-system-base-full} 9.2 and inherits all of its hardware certifications and requirements.
```
Comment (mike-nguyen) [inline modules/installation-machine-requirements.adoc:97]: LGTM here

File: modules/installation-machine-requirements.adoc (hunk 6 above)
Comment (mike-nguyen) [inline modules/installation-machine-requirements.adoc:90]: I can't comment on bring your own RHEL support

File: modules/installation-minimum-resource-requirements.adoc [hunk 8]
```diff
@@ -158,7 +158,7 @@ endif::ibm-z[]
 ifndef::openshift-origin[]
 |Compute
 ifdef::ibm-z,ibm-power,ibm-cloud-vpc[|{op-system}]
-ifndef::ibm-z,ibm-power,ibm-cloud-vpc[|{op-system}, {op-system-base} 8.4, or {op-system-base} 8.5 ^[3]^]
+ifndef::ibm-z,ibm-power,ibm-cloud-vpc[|{op-system}, {op-system-base} 8.6, or {op-system-base} 8.7 ^[3]^]
```
Comment (mike-nguyen) [inline modules/installation-minimum-resource-requirements.adoc:161]: I'm not the correct person to review the bring your own RHEL.

**Review by mike-nguyen — COMMENTED** [review]
I can only LGTM the pieces that are RHCOS specific. I am not the right person to comment on the bring your own RHEL support.

File: modules/installation-machine-requirements.adoc (hunk 6 above)
Comment (marrusl) [reply modules/installation-machine-requirements.adoc:90]: LGTM. Thanks!

**Comment by stevsmit** [comment]
lgtm

**Pull request merged**